app/apis/api_admin.py

```diff
@@ -0,0 +1,139 @@
+from models.application_user_role import ApplicationUserRole, Role
+from models.user import User
+
+admin_info_namespace = Namespace('admin', description='Admin Endpoint.')
+success_or_not = admin_info_namespace.model('Success', {
+    'status': fields.Boolean(
+        required=True
+    ),
+    'message': fields.String(
+        required=True
+    )
+})
+user_info = admin_info_namespace.model('User', {
+    'user_uid': fields.String(required=True),
+    'user_name': fields.String(required=True)
+})
+role_info = admin_info_namespace.model('Role', {
+    'role': fields.String(required=True),
+    'user': fields.Nested(user_info, required=True)
+})
+
+
+@admin_info_namespace.route('/users')
+class ApiApplicationUsers(Resource):
+    @admin_info_namespace.marshal_list_with(user_info)
+    def get(self):
+        return User.query.all()
+
+
+def check_role(fn):
+    @wraps(fn)
+    def wrapper(*args, **kwargs):
+        application_id = kwargs.get('application_id')
+        user_id = get_jwt_identity()
+        role = db.session.query(ApplicationUserRole).filter(
+            ApplicationUserRole.application_id == application_id,
+            ApplicationUserRole.user_id == user_id).one_or_none()
+        if role is None:
+            # applications which don't have users are also accesssible as owner
+            roles = db.session.query(ApplicationUserRole).filter(
+                ApplicationUserRole.application_id == application_id).count()
+            if roles == 0:
+                pass
+            else:
+                raise ApplicationUserRoleException
+        elif role.role != Role.owner:
+            raise ApplicationUserRoleException
+        return fn(*args, **kwargs)
+    return wrapper
+
+
+@admin_info_namespace.route('/<int:application_id>/acl')
+class ApiApplicationIdACL(Resource):
+    method_decorators = [check_role]
+
+    save_acl_parser = reqparse.RequestParser()
+    save_acl_parser.add_argument('uid', type=str, required=True, location='form')
+    save_acl_parser.add_argument('role', type=str, required=True, location='form')
+
+    delete_acl_parser = reqparse.RequestParser()
+    delete_acl_parser.add_argument('uid', type=str, required=True, location='form')
+
+    @admin_info_namespace.marshal_list_with(role_info)
+    def get(self, application_id):
+        roles = ApplicationUserRole.query.filter_by(application_id=application_id).all()
+        return roles
+
+    @admin_info_namespace.marshal_with(success_or_not)
+    @admin_info_namespace.expect(save_acl_parser)
+    def post(self, application_id):
+        args = self.save_acl_parser.parse_args()
+        uid = args['uid']
+        uobj = db.session.query(User).filter(User.user_uid == uid).one_or_none()
+        if uobj is None:
+            return {"status": False}, 400
+
+        roleObj = ApplicationUserRole(
+            application_id=application_id,
+            user_id=uobj.user_id,
+            role=args['role'])
+        db.session.add(roleObj)
```

Comment:
We need to check the existence of `args['role']` in the DB first.

Reply:
Since `application_user_roles.role` column is defined as `Enum`, error will return if `args['role']` is invalid value.
https://github.com/rekcurd/drucker-dashboard/pull/49/files#diff-4c2718014a08dc4cbf367a045d8f0935R28
So we don't have to worry about saving incorrect values.

Reply:
There are two kinds of errors in this case.
1. Invalid insertion error. Role which user specify is not exist in our Enum class.
2. Duplicated insertion error. Role already exist in DB.

If these errors happen, system raises exception. But we cannot understand what kind of error happen inside of the backend when we use a dashboard.